internal/services/cognitive/azure_ai_services_resource.go

```diff
@@ -0,0 +1,757 @@
+			existing, err := client.AccountsGet(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return tf.ImportAsExistsError("azurerm_azure_ai_services", id.ID())
+			}
+
+			networkACLs, subnetIds := expandAzureAIServicesNetworkACLs(model.NetworkACLs)
+
+			// also lock on the Virtual Network ID's since modifications in the networking stack are exclusive
+			virtualNetworkNames := make([]string, 0)
+			for _, v := range subnetIds {
+				subnetId, err := commonids.ParseSubnetID(v)
+				if err != nil {
+					return err
+				}
+				if !utils.SliceContainsValue(virtualNetworkNames, subnetId.VirtualNetworkName) {
+					virtualNetworkNames = append(virtualNetworkNames, subnetId.VirtualNetworkName)
+				}
+			}
+
+			locks.MultipleByName(&virtualNetworkNames, network.VirtualNetworkResourceName)
+			defer locks.UnlockMultipleByName(&virtualNetworkNames, network.VirtualNetworkResourceName)
+
+			props := cognitiveservicesaccounts.Account{
+				Kind:     utils.String("AIServices"),
+				Location: utils.String(azure.NormalizeLocation(model.Location)),
+				Sku: &cognitiveservicesaccounts.Sku{
+					Name: model.SkuName,
+				},
+				Properties: &cognitiveservicesaccounts.AccountProperties{
+					NetworkAcls:                   networkACLs,
+					CustomSubDomainName:           pointer.FromString(model.CustomSubdomainName),
+					AllowedFqdnList:               pointer.To(model.Fqdns),
+					PublicNetworkAccess:           pointer.To(cognitiveservicesaccounts.PublicNetworkAccess(model.PublicNetworkAccess)),
+					RestrictOutboundNetworkAccess: pointer.To(model.OutboundNetworkAccessRestricted),
+					DisableLocalAuth:              pointer.To(!model.LocalAuthorizationEnabled),
+				},
+				Tags: pointer.To(model.Tags),
+			}
+
+			expandIdentity, err := identity.ExpandSystemAndUserAssignedMapFromModel(model.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+			props.Identity = expandIdentity
+
+			future, err := client.AccountsCreate(ctx, id, props)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			if err := future.Poller.PollUntilDone(ctx); err != nil {
+				return fmt.Errorf("waiting for creating of %s: %+v", id, err)
+			}
+
+			customMangedKey, err := expandAzureAIServicesCustomerManagedKey(model.CustomerManagedKey)
+			if err != nil {
+				return fmt.Errorf("expanding `customer_managed_key`: %+v", err)
+			}
+
+			if customMangedKey != nil {
+				props.Properties.Encryption = customMangedKey
+				futureUpdate, err := client.AccountsUpdate(ctx, id, props)
+				if err != nil {
+					return fmt.Errorf("updating %s: %+v", id, err)
+				}
+
+				if err := futureUpdate.Poller.PollUntilDone(ctx); err != nil {
+					return fmt.Errorf("waiting for updating of %s: %+v", id, err)
+				}
+			}
```

Comment:
Also this is caused by the KV HSM encryption.  Create new resource with KV HSM may result in timeout after hours(random failure) but create new resource then update it will finished soon, update is more stable

Reply:
Have shared this with the service team but unfortunately no response